Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py

```diff
@@ -474,28 +536,61 @@ def get_indicators(indicators: str) -> list:
     return []
 
 
+def push_iocs(client, iocs, path='tim_insert_jsons/'):
+    path = 'tim_insert_jsons/'
+    validation_errors: list = []
+    demisto.info(f"pushing IOCs to XDR: pushing {len(iocs)} IOCs to the {path} endpoint")
+    for i, single_batch_iocs in enumerate(batch_iocs(iocs, batch_size=MAX_INDICATORS_TO_SYNC)):
+        demisto.debug(f'pushing IOCs to XDR: batch #{i} with {len(single_batch_iocs)} IOCs')
+        requests_kwargs: dict = get_requests_kwargs(_json=list(
+            map(demisto_ioc_to_xdr, single_batch_iocs)), validate=True)
+        response = client.http_request(url_suffix=path, requests_kwargs=requests_kwargs)
+        validation_errors.extend(response.get('reply', {}).get('validation_errors'))
+    return validation_errors
+
+
+def push_indicators_to_xdr_request(client, indicators, ):
+    path = 'tim_insert_jsons/'
+    demisto.debug(f"pushing IOCs to XDR: pushing {len(indicators)} IOCs to the {path} endpoint")
+    requests_kwargs: dict = get_requests_kwargs(_json=indicators, validate=True)
+    response = client.http_request(url_suffix=path, requests_kwargs=requests_kwargs)
+    if response.get('reply', {}).get('success') is not True:
+        raise DemistoException(f"Response status was not success, {response=}")
+    return response
+
+
 def tim_insert_jsons(client: Client):
-    # takes our changes and pushes to XDR
+    # Retrieve iocs changes from xsoar and pushes to XDR
     indicators = demisto.args().get('indicator', '')
+    validation_errors: list = []
+    # If tim_insert_jsons is called from xdr-iocs-push is called
     if indicators:
         demisto.info(f"pushing IOCs to XDR: querying with input {indicators}")
         iocs = get_indicators(indicators)
+        if iocs:
+            validation_errors = push_iocs(client, iocs)
```

Comment:
we may need to consider logging validation errors before entering the eWhile True loop.

Reply:
added